Lib/python/pyhead.swg

```diff
@@ -25,6 +25,37 @@
 #  define SWIG_Python_str_FromFormat PyString_FromFormat
 #endif
 
+/*
+  Py_TYPE and related functions are officially part of the limited API only
+  since 3.11, and while we don't actually break ABI compatibility by using
+  them even with the earlier versions because they are implemented as inline
+  functions in Python headers, doing this does provoke an audit failure from
+  abi3audit tool, so avoid using them if we target anything earlier than 3.11.
+ */
+#if defined(Py_LIMITED_API) && Py_LIMITED_API+0 < 0x030B0000
```

Comment:
Actually this could, and probably should, be simplified to just
```suggestion
/*
  Py_TYPE and related functions are officially part of the limited API only
  since 3.14, so avoid using them if we target anything earlier than this.
 */
#if defined(Py_LIMITED_API) && Py_LIMITED_API+0 < 0x030E0000
```
because while we may use them with 3.11, 3.12 and 3.13 without breaking the ABI, this still results in abi3audit error.

I don't know if I should push this change or if you can do it when merging? This won't affect our CI anyhow, as it only uses `PY_ABI_VER==3.4`.

Reply:
maybe its better to avoid any warning, as I definetely want to add a CI job for python>=3.11 when I try to fix https://github.com/swig/swig/issues/3059